framework/core/Project/Mockito.pm

```diff
@@ -73,17 +73,17 @@ sub _post_checkout {
 
     # Change Url to Gradle distribution
     my $prop = "$work_dir/gradle/wrapper/gradle-wrapper.properties";
-    my $lib_dir = "$LIB_DIR/build_systems/gradle";
+    my $lib_dir = "$BUILD_SYSTEMS_LIB_DIR/gradle/dists";
 
     # Read existing Gradle properties file, if it exists
     open(PROP, "<$prop") or return;
     my @tmp;
     while (<PROP>) {
-        if (/(distributionUrl=).*\/(gradle-2.*)/) {
-            s/(distributionUrl=).*\/(gradle-.*)/$1file\\:$lib_dir\/gradle-2.2.1-all.zip/g;
-        } else {
-            s/(distributionUrl=).*\/(gradle-.*)/$1file\\:$lib_dir\/gradle-1.12-bin.zip/g;
-        }
+        # Replace the online url, e.g.,
+        # https\://services.gradle.org/distributions/gradle-2.4-all.zip
+        # with a local one, e.g.,
+        # file\\:Constants::$BUILD_SYSTEMS_LIB_DIR/gradle/dists/gradle-2.4-all.zip
+        s/(distributionUrl=).*\/(gradle-.*)/$1file\\:$lib_dir\/$2/g;
```

Comment:
I am inclined to stick with the minimal subset of distributions. That said, we should extract the hard-coded mapping of which project version requires what gradle distribution. What about generating such a mapping in the `get_gradle_dependencies.pl`script?

I think we should leave the current logic to get this PR (which is about the dependencies, not the distributions) merged and open an enhancement issue for a better handling of gradle distributions.